packages/api/src/email-campaign/email-campaign.resolver.ts

```diff
@@ -49,11 +54,21 @@ export function createEmailCampaignsResolver({
         }
 
         @Query(() => PaginatedEmailCampaigns)
-        async emailCampaigns(@Args() { search, filter, sort, offset, limit, scope }: EmailCampaignsArgs): Promise<PaginatedEmailCampaigns> {
+        async emailCampaigns(
+            @Args() { search, filter, sort, offset, limit, scope }: EmailCampaignsArgs,
+            @Info() info: GraphQLResolveInfo,
+        ): Promise<PaginatedEmailCampaigns> {
             const where = this.campaignsService.getFindCondition({ search, filter });
             where.scope = scope;
 
-            const options: FindOptions<EmailCampaignInterface> = { offset, limit };
+            const fields = extractGraphqlFields(info, { root: "nodes" });
+            const populate: string[] = [];
+            if (fields.includes("targetGroup")) {
+                populate.push("targetGroup");
+            }
+
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            const options: FindOptions<EmailCampaignInterface, any> = { offset, limit, populate };
```

Comment:
is the api generator defining this as any or is this custom?
Is it because of the populate option?

Reply:
Yes and yes 👍🏻